core/HmmModel.cc

```diff
@@ -81,9 +105,19 @@ void HmmModel2::PersistParams(string &prefix) {
 }
 
 // builds the lattice of all possible label sequences
-void HmmModel2::BuildThetaGammaFst(vector<int64_t> &x, VectorFst<FstUtils::LogArc> &fst) {
+void HmmModel2::BuildThetaGammaFst(vector<int64_t> &x, VectorFst<FstUtils::LogArc> &fst, unsigned sentId) {
   // arcs represent a particular choice of y_i at time step i
   // arc weights are - log \theta_{x_i|y_i} - log \gamma_{y_i|y_{i-1}}
+    
+    if(sentId==55665566) {
+        assert(learningInfo->neuralRepFilename.empty());    
+    }
+    
+    vector<Eigen::VectorNeural> neural;
+    if(! learningInfo->neuralRepFilename.empty()) {
+        neural = GetNeuralSequence(sentId);
```

Comment:
as discussed in the code review, we are going to store the embeddings per word type, rather than per sentence id. this will reduce the memory requirements because some words are repeated many times in different sentence ids. we will have the utility function return a map from word string to eigen vector, and use that to populate a map<int64_t, EigenVector> as a member of HmmModel2 class.